feat(localnet) - add localnet log aggregation tool

mur-me opened the pull request:
## Issue

I've covered localnet deployment with the same tool set as higher networks - Promtail-Loki-Grafana.

![image](https://github.com/user-attachments/assets/e2d0eadb-d2d5-4cca-95d4-2a456fafd0c7)

What was done: 
* 3 Makefile commands to start and stop log aggregation - `debug-start-log` and `debug-stop-log`, `debug-restart-log`
* these 2 new Makefile are just are wrappers between our localnet logs location, promtail-grafana-loki configs and the docker-compose file
* WARNING - grafana runs in the anon access, just in case - **DON'T RUN IT AGAINST PRODUCTION**

Details:
*  test/logs_aggregator/start_log_aggregator.sh
  * go to `test/logs_aggregator/` folder
  * find the current localnet logs folder
  * put it inside `.env` file in `test/logs_aggregator/.env`
  * this `test/logs_aggregator/.env` is a source of variables for docker-compose
  * start docker-compose detached
  * open the Grafana in browser
* test/logs_aggregator/stop_log_aggregator.sh
  * just cleanup the `test/logs_aggregator/.env`
  * stop and remove docker compose

## Test

Local run:
* make debug-start-log 
```
bash ./test/logs_aggregator/start_log_aggregator.sh
working in /home/uladzislau/go/src/github.com/harmony-one/harmony/test/logs_aggregator
/home/uladzislau/go/src/github.com/harmony-one/harmony/test/logs_aggregator
Current localnet logs are placed into '/home/uladzislau/go/src/github.com/harmony-one/harmony/tmp_log/log-20240916-175304'
CURRENT_SESSION_LOGS='/home/uladzislau/go/src/github.com/harmony-one/harmony/tmp_log/log-20240916-175304'
starting docker compose lor log aggregation
[+] Running 3/3
 ✔ Container logs_aggregator-promtail-1  Started                                                                                              1.2s 
 ✔ Container logs_aggregator-loki-1      Started                                                                                              1.2s 
 ✔ Container logs_aggregator-grafana-1   Started                                                                                              1.3s 
Opening Grafana
```

* make debug-stop-log
``` bash ./test/logs_aggregator/stop_log_aggregator.sh
working in /home/uladzislau/go/src/github.com/harmony-one/harmony/test/logs_aggregator
/home/uladzislau/go/src/github.com/harmony-one/harmony/test/logs_aggregator
[INFO] - stopping log aggregation
[+] Stopping 3/3
 ✔ Container logs_aggregator-loki-1      Stopped                                                                                              2.6s 
 ✔ Container logs_aggregator-promtail-1  Stopped                                                                                              0.6s 
 ✔ Container logs_aggregator-grafana-1   Stopped                                                                                              0.6s 
Going to remove logs_aggregator-promtail-1, logs_aggregator-grafana-1, logs_aggregator-loki-1
[+] Removing 3/0
 ✔ Container logs_aggregator-promtail-1  Removed                                                                                              0.0s 
 ✔ Container logs_aggregator-grafana-1   Removed                                                                                              0.0s 
 ✔ Container logs_aggregator-loki-1      Removed                                                                                              0.0s 
[INFO] - cleanup .env
```
* make debug-stop-log-aggregator against already stopped/not-started localnet it will be failure
* make debug-start-log-aggregator against not-started localnet it will be failure

* make debug-restart-log:
```
bash ./test/logs_aggregator/stop_log_aggregator.sh
working in /home/uladzislau/go/src/github.com/harmony-one/harmony/test/logs_aggregator
/home/uladzislau/go/src/github.com/harmony-one/harmony/test/logs_aggregator
[INFO] - stopping log aggregation
[+] Stopping 3/3
 ✔ Container localnet-promtail  Stopped                                                                                                       0.6s 
 ✔ Container localnet-grafana   Stopped                                                                                                       0.6s 
 ✔ Container localnet-loki      Stopped                                                                                                       1.8s 
Going to remove localnet-grafana, localnet-promtail, localnet-loki
[+] Removing 3/0
 ✔ Container localnet-loki      Removed                                                                                                       0.1s 
 ✔ Container localnet-grafana   Removed                                                                                                       0.1s 
 ✔ Container localnet-promtail  Removed                                                                                                       0.1s 
[INFO] - cleanup .env
bash ./test/logs_aggregator/start_log_aggregator.sh
working in /home/uladzislau/go/src/github.com/harmony-one/harmony/test/logs_aggregator
/home/uladzislau/go/src/github.com/harmony-one/harmony/test/logs_aggregator
Current localnet logs are placed into '/home/uladzislau/go/src/github.com/harmony-one/harmony/tmp_log/log-20240920-122014'
CURRENT_SESSION_LOGS='/home/uladzislau/go/src/github.com/harmony-one/harmony/tmp_log/log-20240920-122014'
starting docker compose lor log aggregation
[+] Running 3/3
 ✔ Container localnet-loki      Started                                                                                                       0.6s 
 ✔ Container localnet-grafana   Started                                                                                                       0.7s 
 ✔ Container localnet-promtail  Started                                                                                                       0.5s 
Opening Grafana
```

### Unit Test Coverage

Before:

```
<!-- copy/paste 'go test -cover' result in the directory you made change -->
```

After:

```
<!-- copy/paste 'go test -cover' result in the directory you made change -->
```

### Test/Run Logs

<!-- links to the test/run log, or copy&paste part of the log if it is too long -->
<!-- or you may just create a [gist](https://gist.github.com/) and link the gist here -->

## Operational Checklist

1. **Does this PR introduce backward-incompatible changes to the on-disk data structure and/or the over-the-wire protocol?**. (If no, skip to question 8.)

    **YES|NO**

2. **Describe the migration plan.**. For each flag epoch, describe what changes take place at the flag epoch, the anticipated interactions between upgraded/non-upgraded nodes, and any special operational considerations for the migration.

3. **Describe how the plan was tested.**

4. **How much minimum baking period after the last flag epoch should we allow on Pangaea before promotion onto mainnet?**

5. **What are the planned flag epoch numbers and their ETAs on Pangaea?**

6. **What are the planned flag epoch numbers and their ETAs on mainnet?**

    Note that this must be enough to cover baking period on Pangaea.

7. **What should node operators know about this planned change?**

8. **Does this PR introduce backward-incompatible changes *NOT* related to on-disk data structure and/or over-the-wire protocol?** (If no, continue to question 11.)

    **YES|NO**

9. **Does the existing `node.sh` continue to work with this change?**

10. **What should node operators know about this change?**

11. **Does this PR introduce significant changes to the operational requirements of the node software, such as >20% increase in CPU, memory, and/or disk usage?**

## TODO

sophoah commented on test/logs_aggregator/start_log_aggregator.sh:
```diff
@@ -0,0 +1,17 @@
+echo "CURRENT_SESSION_LOGS='${path}/${logs}'" > .env
+echo "CURRENT_SESSION_LOGS='${path}/${logs}'"
+echo "starting docker compose lor log aggregation"
+docker-compose up --detach
```
i believe the latest version of docker docker-compose is no longer using `docker-compose` but `docker compose`

mur-me replied:
Hey, `docker-compose` backward compatible, but I agreed to use the new one, less confusion + it will supported in the future releases

mur-me replied:
Backward compatible, but agreed, new release can remove deprecated command

sophoah commented on Makefile:
```diff
@@ -216,3 +216,11 @@ build_localnet_validator:
+
+# start a log aggregation toolset(Promtail->Loki->Grafana)
+# needs docker and docker-compose
+debug-start-log-aggregator:
```
feels a bit long to type what do you think of 
debug-start-logagg (or simpy debug-start-log) 

and similar naming for `stop`

mur-me replied:
agreed

mur-me replied:
+ added these 2 commands to the `help` command

mur-me replied:
+ one more command `debug-restart-log` - just do `debug-restart-log: debug-stop-log debug-start-log`

sophoah commented on test/logs_aggregator/docker-compose.yml:
```diff
@@ -0,0 +1,18 @@
+services:
+  loki:
```
add `container_name` for each service else the container default name can be quite long ie :
-  `  logs_aggregator-loki-1` 
-  `  logs_aggregator-grafana-1` 
-  `  logs_aggregator-promtail-1`

mur-me replied:
gotcha,it will be now:
 ✔ Container localnet-loki      Started                                                                                                       0.6s 
 ✔ Container localnet-grafana   Started                                                                                                       0.7s 
 ✔ Container localnet-promtail  Started

sophoah commented:
hey @mur-me good job on this. I made a few comments and manage to run the stack with my localnet so all great !

sophoah commented:
hey @GheisMohammadi @Frozen please test the stack, this PR is for you guys

mur-me commented:
> hey @mur-me good job on this. I made a few comments, please address. Additionally, I've managed to run the stack with my localnet !

Addressed all the comments, added a few improvements like restart and anon accces to the grafana

sophoah commented on test/logs_aggregator/docker-compose.yml:
```diff
@@ -0,0 +1,25 @@
+    volumes:
+      - ./loki-config.yml:/etc/loki/loki-config.yaml
+  promtail:
+    container_name: localnet-promtail
```
shorter !! hehe
the docker compose service name is enough if you ask me for the container name

mur-me replied:
Done! :D
